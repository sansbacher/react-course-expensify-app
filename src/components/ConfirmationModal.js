import React from 'react'
import Modal from 'react-modal'

// Tell React-Modal what element (our root div) to hide when displaying the modal for assistive readers:
// http://reactcommunity.org/react-modal/accessibility/
if (process.env.NODE_ENV !== 'test') {						// To avoid errors when testing with Jest: https://github.com/reactjs/react-modal/issues/632
	Modal.setAppElement('#app')
}

const ConfirmationModal = props => (
	<Modal
		isOpen={props.showConfirmation}
		contentLabel="Confirm Remove"
		closeTimeoutMS={150}
		className="modal"
	>
		<h3 className="modal__title">Remove Expense?</h3>
		<p className="modal__body">Are you sure you want to remove this Expense:</p>
		<p className="modal__body--message">{props.expenseDescription}</p>
		<div>
			<button className="button" onClick={props.handleRemove}>Remove</button>
			<button className="button button--secondary" onClick={props.handleCancel}>Cancel</button>
		</div>
	</Modal>
)

export default ConfirmationModal;