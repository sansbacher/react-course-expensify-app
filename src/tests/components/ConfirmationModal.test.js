// Jest provides test() as a global. BUT can use this to help VSCode out:
import {expect, test, jest} from '@jest/globals'				// With React/Browser

import React from 'react'
import { shallow } from 'enzyme'
import ConfirmationModal from '../../components/ConfirmationModal'

// jest.spyOn(ConfirmationModal, 'setAppElement').mockImplementation( param => console.log('setAppElement', param))

test('should render ConfirmationModal correctly', () => {
	const wrapper = shallow(
		<ConfirmationModal
			showConfirmation={true}
			expenseDescription="Some dummy expense"
		>
			Some dummy model content
		</ConfirmationModal>
	)
	expect(wrapper).toMatchSnapshot()
})

test('should activate Remove correctly', () => {
	const handleRemove = jest.fn()
	// Not testing the handleCancel callback
	const wrapper = shallow(
		<ConfirmationModal
			showConfirmation={true}
			expenseDescription="Some dummy expense"
			handleRemove={handleRemove}
		>
			Some dummy model content
		</ConfirmationModal>
	)
	wrapper.find('button').at(0).simulate('click')
	expect(handleRemove).toHaveBeenCalled()
})
