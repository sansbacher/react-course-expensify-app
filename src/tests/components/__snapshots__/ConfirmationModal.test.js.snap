// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ConfirmationModal correctly 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal"
  closeTimeoutMS={150}
  contentElement={[Function]}
  contentLabel="Confirm Remove"
  isOpen={true}
  overlayElement={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  preventScroll={false}
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <h3
    className="modal__title"
  >
    Remove Expense?
  </h3>
  <p
    className="modal__body"
  >
    Are you sure you want to remove this Expense:
  </p>
  <p
    className="modal__body--message"
  >
    Some dummy expense
  </p>
  <div>
    <button
      className="button"
    >
      Remove
    </button>
    <button
      className="button button--secondary"
    >
      Cancel
    </button>
  </div>
</Modal>
`;
